### ISSUE TITLE
Library problems

### ISSUE THREAD

**Issue by subogero** [popcornmix/omxplayer#412]
With the latest build I get the following error on an up-to date Minibian (Raspbian without X) system:

    /usr/bin/omxplayer.bin: /lib/arm-linux-gnueabihf/libc.so.6: version `GLIBC_2.17' not found (required by /usr/lib/omxplayer/libavutil.so.54)
    /usr/bin/omxplayer.bin: /usr/lib/arm-linux-gnueabihf/libsmbclient.so.0: no version information available (required by /usr/lib/omxplayer/libavformat.so.56)

By latest build I mean

    omxplayer_0.3.6~git20160102~f544084_armhf.deb

from the http://omxplayer.sconde.net build system.

The previous build works perfectly:

    omxplayer_0.3.6~git20150627~843744e_armhf.deb

**Comment by jehutting** [comment]
Hmmm...is this an OMXPlayer problem or a problem of your Minibian distribution?

On Jessie
> $ /lib/arm-linux-gnueabihf/libc.so.6 --version
> GNU C Library (Debian GLIBC 2.19-18+deb8u1) stable release version 2.19, by Roland McGrath et al.

>$ ls -l /usr/lib/omxplayer/libavu*
lrwxrwxrwx 1 root root     22 jan  2 14:46 /usr/lib/omxplayer/libavutil.so -> libavutil.so.54.31.100
lrwxrwxrwx 1 root root     22 jan  2 14:46 /usr/lib/omxplayer/libavutil.so.54 -> libavutil.so.54.31.100
-rwxr-xr-x 1 root root 373472 jan  2 14:46 /usr/lib/omxplayer/libavutil.so.54.31.100


On Wheezy
>$ /lib/arm-linux-gnueabihf/libc.so.6 --version
GNU C Library (Debian EGLIBC 2.13-38+rpi2+deb7u8) stable release version 2.13, by Roland McGrath et al.

I do not know how/where this specific need (of omxplayer) for 2.17 is set/defined/configured.

**Comment by anta40** [comment]
Yep, I'm having the same problem.
Have no idea what the workaround is....

**Comment by jehutting** [comment]
Maybe @skgsergio can help...
Are you building the omxplayer package with libc version 2.17, and when people running at wheezy -where it is stuck on 2.13- getting this issue? As I already said. I don't know how the versioning works.

@anta40 on which os do you run? Could you run `cat /etc/os-release` and above version commands? Thanks in advance.

**Comment by skgsergio** [comment]
I am building

**Comment by skgsergio** [comment]
I am building against the Jessie glibc since I updated my build system to it in January... Someone contacted me with some problems with Jessie and my builds so I decided to update... 

I could rollback the system but maybe then Jessie users will have problems (and I don't have much time for doing tests) so... I don't know, my build machine have little resources and takes it time to build it so building for Jessie and Whezzy is not the best scenario for me but to be fair now omxplayer update less so let me think (and until Monday I can't do something).

**Comment by jehutting** [comment]
Thanks @skgsergio for clarification.

I installled on my Wheezy RPI omxplayer_0.3.6~git20160102~f544084_armhf.deb.
Now I get the libc and libsmbclient errors too.

So the problem is caused by the lib versions the packages are built with!?

**Comment by skgsergio** [comment]
@jehutting Yes, since jessie have greater versions it fails.

I'm rolling back my system to wheezy (since it seems most of the users still use wheezy) and upload a new build ASAP (about 5h from now I have to go out now, but I scripted all so if it goes ok we only will have to wait for me to upload the file).

Then I'll check the problem a user reported me with jessie (to be honest I didn't pay much attention to it and just updated the buildenv...) and if it's needed I'll make separate builds for wheezy and jessie...

**Comment by anta40** [comment]
Hi Sergio, 
If the easiest solution is upgrading to Jessie than I guess it's fine. As long as my Python script (which relies on omxplayer) runs properly... :D


Sent from SM-N910H

-------- Original message --------

**Comment by skgsergio** [comment]
@anta40 I'm going back to wheezy builds as first and quick solution and I'll later check the jessie compatibility since most people seems to use wheezy.

**Comment by popcornmix** [comment]
Wheezy is getting no updates form raspberry pi. Jessie is the way forward, so Jessie usage will only increase.

**Comment by skgsergio** [comment]
yes @popcornmix that should be the trend but from this ticket and some emails I received seems that the people is not updating their systems yet.

**Comment by skgsergio** [comment]
Btw, @anta40, @jehutting and @subogero built latest version for wheezy: http://omxplayer.sconde.net/builds/omxplayer_0.3.7~git20160206~cb91001_armhf.deb

**Comment by jehutting** [comment]
Thanks @skgsergio! Installed the deb and the errors are not shown anymore (on the wheezy distro :).

**Comment by anta40** [comment]
Hi @skgsergio 

The latest build installed fine on my Wheezy. The looping problem still persists though.
I tried this:
`omxplayer --loop Particle_25_mepet_max.mp4`

The video only played once :(
Just in case you want to try it yourself, [this is the link of the video](http://lab.anta40.net/vids/Particle_25_mepet_max.mp4):

**Comment by jehutting** [comment]
@subogero So this is an issue for the Minibian distribution?

**Comment by jehutting** [comment]
Ping @subogero

**Comment by jehutting** [comment]
Ping... Ping... @subogero

**Comment by Ruffio** [comment]
@popcornmix I think this can be closed
